packages/libs/deer/src/error.rs

```diff
@@ -1,15 +1,196 @@
+//! # Error Structure
```

Comment:
Is this similar to `serde::de::Error`?

Reply:
Yes, we're essentially replacing the `serde::de::Error` with an `error-stack` based approach, that allows greater flexibility.